Sources/SwiftToolchain/Toolchain.swift

```diff
@@ -118,8 +121,30 @@ public final class Toolchain {
     }
   }
 
+  private func inferManifestDirectory() throws -> AbsolutePath {
+    guard package != nil, var cwd = fileSystem.currentWorkingDirectory else {
+      throw ToolchainError.missingPackage
+    }
+
+    // FIXME: this won't work on Windows
+    let root = AbsolutePath("/")
+
+    repeat {
+      guard !fileSystem.isFile(cwd.appending(component: "Package.swift")) else {
+        return cwd
+      }
+
+      // `parentDirectory` just returns `self` if it's `root`
+      cwd = cwd.parentDirectory
+    } while cwd != root
```

Comment:
It seems that `AbsolutePath` has a property `isRoot` to check whether it's a root directory.
```suggestion
    } while !cwd.isRoot
```

Reply:
Great point, I initially wasn't sure if this property is public at all.